torch/onnx/symbolic.py

```diff
@@ -859,10 +859,22 @@ def upsample_bilinear2d(g, input, output_size, align_corners):
                 mode_s="linear")
 
 
-def wrap_logical_op_with_cast_to_uint8(func):
-    def wrap_with_cast(g, input, other):
-        return g.op("Cast", func(g, input, other), to_i=cast_pytorch_to_onnx['Byte'])
-    return wrap_with_cast
+def wrap_logical_op_with_cast_to(to_type):
+    def decorator(fn):
+        def wrap_with_cast(g, input, other):
+            return g.op("Cast", fn(g, input, other), to_i=cast_pytorch_to_onnx[to_type])
+        return wrap_with_cast
+    return decorator
+
+
+def wrap_logical_op_with_cast_to_and_from(to_type):
+    def decorator(fn):
+        def wrap_with_cast(g, input, other):
+            cast_func = globals()['_cast_{}'.format(to_type)]
```

Comment:
is globals() necessary here?

Reply:
rename `cast_func` to `to_cast_func`?

Reply:
done, globals() is for calling methods created here https://github.com/pytorch/pytorch/blob/270b5814ea0cbcbe5e50b574ff535d8605112b0e/torch/onnx/symbolic.py#L1303